sync/src/main/java/tech/pegasys/artemis/sync/PeerSync.java

```diff
@@ -113,6 +121,7 @@ public void stop() {
                   startSlot,
                   peer.getId());
               final UnsignedLong nextSlot = startSlot.plus(count);
+              currentSlot = nextSlot; // might end up at highest + 1 but by then syncing is finished
```

Comment:
The `currentSlot` is already available from `ChainStorageClient.getBestSlot()` and that will correctly show the last imported block, not just the last downloaded block.

`highestSlot` probably should be calculated from `SyncManager` as well since it's the highest slot of any peer, not just the slot of the peer we're syncing to at the moment.

And then at that point you'd only have a single variable being updated here, so it could be `volatile` and remove all the `synchronized`.

Reply:
fixed